client/src/index.jsx

```diff
@@ -117,18 +149,39 @@ class App extends React.Component {
     let targetValue = e.target.value;
     const currentValues = Object.assign({}, this.state.currentValues);
 
+    if(targetValue === ''){
+      targetValue = 0;
+    }
+
     if (targetName !== 'loan_program') {
       targetValue = parseFloat(targetValue);
     }
 
+    if(targetValue.toString().length > 9){
+      targetValue = 0;
+      alert('invalid input');
+    }
+
+    if(targetName !== 'home_price' &&  targetValue > this.state.currentValues.home_price){
+      targetValue = 0;
+      alert('invalid input');
+    }
+
+    if(targetName === 'home_price'){
+      currentValues.down_payment_amount = Math.floor(targetValue*(this.state.currentValues.down_payment_percentage/100));
+    }
     if (targetName === 'down_payment_percentage') {
-      currentValues.down_payment_amount = Math.floor(this.state.currentValues.home_price * targetValue / 100);
+        currentValues.down_payment_amount = Math.floor(this.state.currentValues.home_price * targetValue / 100);
     } else if (targetName === 'down_payment_amount') {
-      currentValues.down_payment_percentage = ((targetValue / this.state.currentValues.home_price) * 100).toFixed(0);
+      if(currentValues.down_payment_amount = 0){
+        currentValues.down_payment_percentage = 0;
+      }else{
```

Comment:
space between parens
```suggestion
      } else {
```